app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+/**
+ * This view is the drawable area of the app.
+ * drawing on this location will draw to a bitmap which is loaded
+ * onBitmapDrawn saves the bitmap to a different location as needed
+ * attemptBitmapUpdate will update the bitmap to another one if provided to it
+ * At the start of onDraw, if the bitmap is null, it looks to update to a bitmap
+ * if no bitmap is found, it will generate a blank one
+ * After being interacted with, it will save the current bitmap
+ */
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    /**
+     * This lambda takes in a bitmap and returns nothing
+     * This is being used to save the bitmap to a different location
+     */
+    var onBitmapDrawn: ((Bitmap?) -> Unit)? = null
+
+    /**
+     * This lambda will attempt to update the bitmap to another bitmap
+     * this is done if we have a null bitmap at the start of drawing
+     */
+    var attemptBitmapUpdate: (() -> Unit)? = null
```

Comment:
Why does this even need to be a lambda? Why not just `setBitmap` that sets `userDrawnBitmap` to whatever they bring in

Reply:
Introduced viewModel.currentBitmap?.let { drawableArea.setBitmap(it) } in the Main Activity after I initialize the view. Instead of using a callback lambda, it will now attempt to set the viewModels current bitmap as the bitmap of the drawable area. Tested locally and found that it does correctly update the bitmap on orientation changes.

As a result I made the updateViews function private in the viewModel since nothing needs to call on it directly anymore outside of the viewModel.